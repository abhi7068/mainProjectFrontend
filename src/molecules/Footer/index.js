import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import StyledLink from '../../atoms/StyledLink';
import { useMediaQuery } from '@material-ui/core';
import MobileFooter from '../../atoms/MobileFooter';
const useStyles = makeStyles((theme) => ({
  toolbar: {
    borderBottom: `1px solid ${theme.palette.divider}`,
  },
  toolbarTitle: {
    flex: 1,
  },
  toolbarSecondary: {
    justifyContent: 'space-around',
    position: 'fixed',
    bottom: '0',
    background: '#E40046 ',
    height: '70',
    width: '100%',
    overflowX: 'auto',
  },
  toolbarLink: {
    // padding: theme.spacing(1),
    flexShrink: 0,
  },
  coloring: {
    background: '#E40046',
  },
}));
function Footer() {
  const mobileView = useMediaQuery('(max-width:460px)');
  const bigscreen = useMediaQuery('(min-width:460px)');
  const classes = useStyles();
  return (
    <div>
      {bigscreen && (
        <div className={classes.coloring}>
          <Toolbar
            component="nav"
            variant="dense"
            className={classes.toolbarSecondary}
          >
            <StyledLink to="/mail" innerText="support@yoyogift.com" />
            <StyledLink to="/aboutus" innerText="About Us" />
            <StyledLink to="/contactus" innerText="Contact Us" />
          </Toolbar>
        </div>
      )}
      {mobileView && <MobileFooter />}
    </div>
  );
}
export default Footer;
