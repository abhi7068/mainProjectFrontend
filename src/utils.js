import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
