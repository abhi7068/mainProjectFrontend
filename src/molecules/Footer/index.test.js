import React from 'react';
import { render, cleanup } from '../../utils';
import Footer from './index';

afterEach(cleanup);

it('creatingsnapshot', () => {
  const { asFragment } = render(<Footer />);
  expect(asFragment()).toMatchSnapshot();
});
