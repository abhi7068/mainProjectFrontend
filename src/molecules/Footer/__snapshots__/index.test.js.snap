// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creatingsnapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
